package/MDAnalysis/core/_get_readers.py

```diff
@@ -176,6 +177,14 @@ def get_writer_for(filename, format=None, multiframe=None):
                 None)
         else:
             format = util.check_compressed_format(root, ext)
+        
+    if format == '':
```

Comment:
Under which circumstances can this happen? In particular, can [`util.check_compressed_format(root, ext)`](https://github.com/MDAnalysis/mdanalysis/blob/3b0bdcebd9f087d8b8497815616e53003de6ab2a/package/MDAnalysis/lib/util.py#L933) ever return `""`? — I think the answer is yes but I didn't try it and it's not really documented.

Reply:
Do you prefer `get_ext()` to raise an error if `ext==''` ? In this way we would not need to rewrite the error message and we could actually use the except block `except (TypeError, AttributeError):`.